corehq/apps/data_interfaces/forms.py

```diff
@@ -550,18 +590,33 @@ def clean_property_match_definitions(self):
                     'property_value': property_value,
                     'match_type': match_type,
                 })
-
         return result
 
     def clean_filter_on_closed_parent(self):
         return true_or_false(self.cleaned_data.get('filter_on_closed_parent'))
 
-    def save_criteria(self, rule):
+    def clean_location_filter_definition(self):
+        value = self.cleaned_data.get('location_filter_definition')
+        try:
+            value = json.loads(value)
+        except (TypeError, ValueError):
+            self._json_fail_hard()
+
+        if value:
+            location_def = value[0]
+            if not location_def['include_child_locations']:
+                location_def['include_child_locations'] = False
+
+            return location_def
+        return ''
+
+    def save_criteria(self, rule, save_meta=True):
         with transaction.atomic():
-            rule.case_type = self.cleaned_data['case_type']
-            rule.filter_on_server_modified = self.cleaned_data['filter_on_server_modified']
-            rule.server_modified_boundary = self.cleaned_data['server_modified_boundary']
-            rule.save()
+            if save_meta:
```

Comment:
The reason I added this `save_meta` here (not sold on the name, any suggestions welcome) is because `save_criteria` would overwrite the `case_type` that is already set on the rule [here](https://github.com/dimagi/commcare-hq/blob/03e7f7035235a667d5095c8af6caea9ac4ab79b2/corehq/apps/data_interfaces/views.py#L1011) (and I don't care about `filter_on_server_modified` and `server_modified_boundary`).

Reply:
Although, thinking of it now I could probably also just have done something like 
```suggestion
            if self.cleaned_data.get('case_type'):
                rule.case_type = self.cleaned_data['case_type']
```
For the `DedupeCaseFilterForm`, the `case_type` is not being set (that's why I overwrite the required flag [here](https://github.com/dimagi/commcare-hq/blob/6a5d5f1a6d8a48f6a8e2bd28eb8f78f391046bde/corehq/apps/data_interfaces/forms.py#L868)), since the `case_type` already exists on the rule.